source/connection-string/connection-string-spec.md

```diff
@@ -216,13 +216,24 @@ The values in connection options MUST be URL decoded by the parser. The values c
   ```
 
 - Key value pairs: A value that represents one or more key and value pairs. Multiple key value pairs are delimited by a
-  comma (","). The key is everything up to the first colon sign (":") and the value is everything afterwards. If any
-  keys or values containing a comma (",") or a colon (":") they must be URL encoded. For example:
+  comma (","). The key is everything up to the first colon sign (":") and the value is everything afterwards. Reserved
+  characters such as ':' must be escaped according to [RFC 2396](https://www.rfc-editor.org/rfc/rfc2396)
```

Comment:
I'm not sure that we need to change anything in this spec. The second (or third...) colon must already be treated as part of the value, since the spec already says:

> up to the first colon sign and the value is everything afterwards

The original wording is problematic because it confuses restrictions on users (doc requirements) with driver requirements, and does not specify enough restricted characters, but I don't think we should correct this in this PR, but as part of future work to clarify this spec. 

Likewise, I don't think we should perform other clarifications in this PR, since the formulation is not obvious. For example, I don't think we should refer to the URI RFC, and should instead just say "percent-decoded" (we do not have URI components since connection strings are not URIs, and even if we did, we would still need to, according to the RFC, define the "set of characters actually reserved" within each particular component; note also that `%` is not in the list in section 2.2, though obviously it needs to be encoded by users).

Reply:
Updated